Sources/OutputStream+Write.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  OutputStream+Write.swift
+//  FTAPIKit
+//
+//  Created by Matěj Kašpar Jirásek on 27/02/2019.
+//  Copyright © 2019 FUNTASTY Digital s.r.o. All rights reserved.
+//
+
+import Foundation
+
+extension OutputStream {
+    private static let streamBufferSize = 1024
+
+    func write(inputStream: InputStream) throws {
+        inputStream.open()
+        defer { inputStream.close() }
+
+        while inputStream.hasBytesAvailable {
+            var buffer = [UInt8](repeating: 0, count: OutputStream.streamBufferSize)
+            let bytesRead = inputStream.read(&buffer, maxLength: OutputStream.streamBufferSize)
+
+            if let streamError = inputStream.streamError {
+                throw streamError
+            }
+
+            if bytesRead > 0 {
+                if buffer.count != bytesRead {
+                    buffer = Array(buffer[0..<bytesRead])
+                }
+                try write(buffer: &buffer)
+            } else {
+                break
+            }
+        }
+    }
+
+    func write(data: Data) throws {
+        var buffer = [UInt8](repeating: 0, count: data.count)
+        data.copyBytes(to: &buffer, count: data.count)
+
+        return try write(buffer: &buffer)
+    }
+
+    func write(buffer: inout [UInt8]) throws {
+        var bytesToWrite = buffer.count
+
+        while bytesToWrite > 0, hasSpaceAvailable {
```

Comment:
If hasSpaceAvailable is false and there are still bytes to write, shouldn’t the write function throw an error?

Reply:
Good point, we might have found a bug in Alamofire since I was a bit inspired by their code. 🙈

Reply:
I will leave it like this, since it works now and the code is now invented by me. So, reworking this might introduce some more issues.